playbooks/roles/ufw/tasks/main.yml

```diff
@@ -0,0 +1,72 @@
+---
+- name: Ensure UFW allows SSH (22/TCP)
+  ufw:
+    rule="allow"
+    name="OpenSSH"
+
+- name: Ensure UFW is enabled and denies by default
+  ufw:
+    state="enabled"
+    policy="deny"
+    direction="incoming"
+
+- name: Ensure UFW allows L2TP / IPSec (500/UDP)
+  ufw:
+    to_port="500"
+    proto="udp"
+    rule="allow"
+
+- name: Ensure UFW allows L2TP / IPSec (1701/UDP)
+  ufw:
+    to_port="1701"
+    proto="udp"
+    rule="allow"
+
+- name: Ensure UFW allows L2TP / IPSec (1701/UDP)
+  ufw:
+    to_port="4500"
+    proto="udp"
+    rule="allow"
+
+- name: Ensure UFW allows HTTPS (443/TCP)
+  ufw:
+    to_port="443"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows OpenVPN DNS (53/UDP) to 10.8.0.1
+  ufw:
+    to_port="53"
+    proto="udp"
+    rule="allow"
+    from_ip="10.8.0.1"
+
+- name: Ensure UFW allows OpenVPN (636/TCP)
+  ufw:
+    to_port="636"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows Shadowsocks (8530/TCP)
+  ufw:
+    to_port="8530"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows stunnel (993/TCP)
+  ufw:
+    to_port="993"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows TOR bridge (8443/TCP)
+  ufw:
+    to_port="8443"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows TOR bfs3 pluggable transport (57366/TCP)
```

Comment:
The port numbers for the obfsproxy pluggable transports are randomized and will be different on every server. The tor-bridge role sets a fact called `{{ tor_obfs3_port }}` that can be used here.